AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/hierarchical-state-machine-swiftTests/StateMachineTests.swift

```diff
@@ -6,32 +6,22 @@
 //
 
 import XCTest
-@testable import AWSCognitoAuthPlugin
+@testable import HSM
 
 class StateMachineTests: XCTestCase {
 
-    func testDefaultState() {
+    func testDefaultState() async {
         let testMachine = CounterStateMachine.logging()
-        let notified = expectation(description: "notified")
-        testMachine.getCurrentState {
-            XCTAssertEqual($0.value, 0)
-            notified.fulfill()
-        }
-
-        waitForExpectations(timeout: 0.1)
+        let state = await testMachine.getCurrentState
+        XCTAssertEqual(state.value, 0)
     }
 
-    func testBasicReceive() {
+    func testBasicReceive() async  {
         let testMachine = CounterStateMachine.logging()
         let increment = Counter.Event(id: "1", eventType: .increment)
-        testMachine.send(increment)
-        let notified = expectation(description: "notified")
-        testMachine.getCurrentState {
-            XCTAssertEqual($0.value, 1)
-            notified.fulfill()
-        }
-
-        waitForExpectations(timeout: 0.1)
+        await testMachine.send(increment)
+        let state = await testMachine.getCurrentState
```

Comment:
Something that falls naturally out of the async `send` and the fact that we run event processing to completion is that we could now return the new state from the `send` method: `let state = await testMachine.send(increment)`. This would be a change to the API contract, but would reduce boilerplate by removing the `send/currentState` dance. Thoughts?

Reply:
I am not sure whether it is really helpful in real case scenarios where the immediate state change is not relevant. Most of the use cases, we need to listen to state changes via listener and act on it.